frontend/exporter/src/constant_utils.rs

```diff
@@ -409,26 +412,29 @@ impl<'tcx, S: UnderOwnerState<'tcx>> SInto<S, ConstantExpr> for ty::Const<'tcx>
     fn sinto(&self, s: &S) -> ConstantExpr {
         use rustc_middle::query::Key;
         let span = self.default_span(s.base().tcx);
-        let kind = match self.kind() {
-            ty::ConstKind::Param(p) => ConstantExprKind::ConstRef { id: p.sinto(s) },
+        match self.kind() {
+            ty::ConstKind::Param(p) => {
+                let kind = ConstantExprKind::ConstRef { id: p.sinto(s) };
+                let tcx = s.base().tcx;
+                let param_env = tcx.param_env(s.owner_id());
+                let ty = p.find_ty_from_env(param_env);
+                kind.decorate(ty.sinto(s), span.sinto(s))
+            }
             ty::ConstKind::Infer(..) => fatal!(s[span], "ty::ConstKind::Infer node? {:#?}", self),
 
-            ty::ConstKind::Unevaluated(ucv) => match self.translate_uneval(s, ucv) {
+            ty::ConstKind::Unevaluated(ucv) => match self.translate_uneval(s, ucv, span) {
                 TranslateUnevalRes::EvaluatedConstant(c) => return c.sinto(s),
                 TranslateUnevalRes::GlobalName(c) => c,
             },
-            ty::ConstKind::Value(valtree) => {
-                return valtree_to_constant_expr(s, valtree, self.ty(), span)
-            }
+            ty::ConstKind::Value(ty, valtree) => valtree_to_constant_expr(s, valtree, ty, span),
             ty::ConstKind::Error(_) => fatal!(s[span], "ty::ConstKind::Error"),
             ty::ConstKind::Expr(e) => fatal!(s[span], "ty::ConstKind::Expr {:#?}", e),
 
             ty::ConstKind::Bound(i, bound) => {
-                supposely_unreachable_fatal!(s[span], "ty::ConstKind::Bound"; {i, bound, self.ty()});
+                supposely_unreachable_fatal!(s[span], "ty::ConstKind::Bound"; {i, bound});
             }
             _ => fatal!(s[span], "unexpected case"),
-        };
-        kind.decorate(self.ty().sinto(s), span.sinto(s))
+        }
```

Comment:
This is the dreadful change caused by https://github.com/rust-lang/rust/pull/125958. It wasn't that hard in the end, thanks to @W95Psp's help.